components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -372,9 +384,134 @@ abstract class AbstractFetchDownloadService : Service() {
         context.unregisterReceiver(broadcastReceiver)
     }
 
+    @VisibleForTesting
+    internal open fun removeDownloadJob(downloadJobState: DownloadJobState) {
+        downloadJobs.remove(downloadJobState.state.id)
+        if (downloadJobs.isEmpty()) {
+            stopForeground(false)
+            NotificationManagerCompat.from(context).cancel(getForegroundId())
+            clearAllDownloadsNotificationsAndJobs()
+            stopSelf()
```

Comment:
Won`t stopping the service already run through `clearAllDownloadsNotificationsAndJobs()` when the service gets to the destroy step? 🤔 I wonder if calling this here an additional times has made this code more complicated since now `clearAllDownloadsNotificationsAndJobs()` has two responsibilities as previously it was only handling clean up when the service ends.

Reply:
You were right, after updating the code and doing the cleanup only when the service get destroyed made the code more easier to understand/maintain 👍